### PR TITLE
Adding ConstitutiveLawsApplication

### PR DESCRIPTION
**Pull request by philbucher**
**Description**
It has been in the discussion forever to create a ConstitutiveLawsApplication. This PR creates an EMPTY application, which we can then populate with the laws

### CODE REVIEW

**Comment by philbucher** [comment]
Can we proceed with this?
I am only adding the empty application

**Comment by rubenzorrilla** [comment]
> Can we proceed with this?
> I am only adding the empty application

I'm fine with it but I don't know why the contact tests give an error @loumalouomega

**Comment by loumalouomega** [comment]
> > Can we proceed with this?
> > I am only adding the empty application
> 
> I'm fine with it but I don't know why the contact tests give an error @loumalouomega

That'something I have pending to fix (it is an MPC bug)

**Comment by philbucher** [comment]
> > Can we proceed with this?
> > I am only adding the empty application
> 
> I'm fine with it but I don't know why the contact tests give an error @loumalouomega

happening only in the nightly build, for some time already => #7380

**Comment by loumalouomega** [comment]
> > > Can we proceed with this?
> > > I am only adding the empty application
> > 
> > 
> > I'm fine with it but I don't know why the contact tests give an error @loumalouomega
> 
> happening only in the nightly build, for some time already => #7380

I added authors (most of current laws are from @AlejandroCornejo )

**Comment by philbucher** [comment]
Guys can we please proceed with this?
Another two months have passed without anything happening...
Can someone approve please?

File: applications/ConstitutiveLawsApplication/custom_python/constitutive_laws_python_application.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Alejandro Cornejo Velazquez
+//                   Riccardo Rossi
+//
+//
```
Comment (rubenzorrilla) [inline applications/ConstitutiveLawsApplication/custom_python/constitutive_laws_python_application.cpp]: Header format

File: applications/ConstitutiveLawsApplication/constitutive_laws_application_variables.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Alejandro Cornejo Velazquez
+//                   Riccardo Rossi
+//
```
Comment (rubenzorrilla) [inline applications/ConstitutiveLawsApplication/constitutive_laws_application_variables.h]: Header format

File: applications/ConstitutiveLawsApplication/constitutive_laws_application_variables.cpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Alejandro Cornejo Velazquez
+//                   Riccardo Rossi
+//
```
Comment (rubenzorrilla) [inline applications/ConstitutiveLawsApplication/constitutive_laws_application_variables.cpp]: Header format

File: applications/ConstitutiveLawsApplication/constitutive_laws_application.h [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Alejandro Cornejo Velazquez
+//                   Riccardo Rossi
+//
```
Comment (rubenzorrilla) [inline applications/ConstitutiveLawsApplication/constitutive_laws_application.h]: Header format

File: applications/ConstitutiveLawsApplication/constitutive_laws_application.cpp [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Alejandro Cornejo Velazquez
+//                   Riccardo Rossi
+//
```
Comment (rubenzorrilla) [inline applications/ConstitutiveLawsApplication/constitutive_laws_application.cpp]: Header format

**Review by rubenzorrilla — COMMENTED** [review]
Just adding the new app. I'll approve after the minor changes.

File: applications/ConstitutiveLawsApplication/constitutive_laws_application.cpp (hunk 5 above)
Comment (philbucher) [reply applications/ConstitutiveLawsApplication/constitutive_laws_application.cpp]: what do you mean?
I only used the app generator, if it is wrong here then we should also correct it there

Comment (philbucher) [reply applications/ConstitutiveLawsApplication/constitutive_laws_application.cpp]: ah the indentation is what you mean right?

**Comment by AlejandroCornejo** [comment]
@philbucher thank you! As soon as I defend my PhD thesis (I hope in 1-2 months?) I'm going to put a lot of efforts in this issue regarding the new app for the CL. I'll actively participate in this for sure.
Regards

**Comment by philbucher** [comment]
> @philbucher thank you! As soon as I defend my PhD thesis (I hope in 1-2 months?) I'm going to put a lot of efforts in this issue regarding the new app for the CL. I'll actively participate in this for sure.
> Regards

sounds good, thx
ma main plan here is to start with the infrastructure, testing and the interaction with StructuralMechanics and other apps, the actual implementation I leave to the experts :)

**Comment by philbucher** [comment]
@rubenzorrilla I applied the changes

File: applications/ConstitutiveLawsApplication/constitutive_laws_application.cpp (hunk 5 above)
Comment (rubenzorrilla) [reply applications/ConstitutiveLawsApplication/constitutive_laws_application.cpp]: Yes. We should probably correct it in there too.

**Review by rubenzorrilla — APPROVED** [review]
Done! Ready for @AlejandroCornejo to fill it up :wink:

**Comment by philbucher** [comment]
thx guys

**Pull request merged**